src/Mutant.php

```diff
@@ -48,6 +49,11 @@ class Mutant
      */
     protected $process;
 
+    /**
+     * @var MutantResult
+     */
+    protected $result;
```

Comment:
You are not mistaken. Mutant has to many responsibilities. Thats why I wonder if moving this logic from humbug command to Mutant is good move.
You decreasing responsibility of Humbug command but Mutant suffers from increasing its responsibility.